src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1696,33 +1696,57 @@ public async Task Precanceled_Throws()
             }
         }
 
-        [Fact]
-        public async Task CanceledDuringOperation_Throws()
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task SendAsync_CanceledDuringOperation_Throws(bool ipv6)
```

Comment:
This test seems highly timing-dependent. 

I think what you are doing here is. you're flooding the kernel send buffer with so much data (64K per send, 1000 sends) that you're assuming at least some of the send tasks won't complete before you call CancelAfter. Am I understanding this correctly?

I don't see any better way to do this really, but I worry we may see intermittent failures here. Thoughts?

At the very least this should be [Outerloop] I think.

Reply:
Also a comment explaining what's intended would be useful.

Reply:
Actually, since this is TCP (I was thinking UDP for some reason) I think what you're doing should be fine. 

That said, it should be [Outerloop] and a comment would be nice.

Reply:
> you're flooding the kernel send buffer with so much data (64K per send, 1000 sends) that you're assuming at least some of the send tasks won't complete before you call CancelAfter

Yes correct. I also tried to change the send buffer size in the hope that this will fragment and slow down the execution of the sends, but on windows this didn't change anything. Maybe I have a bad understanding on how send buffer works.

> but I worry we may see intermittent failures here

I think this can only happen if an environment is so fast, that all the sends complete before the cancellation. I havent seen this yet with the current parameters. Or do you rather worry because of the side effects this may have on other tests running in parallel?